CMakeLists.txt

```diff
@@ -479,6 +479,11 @@ include_directories(SYSTEM ${PROJECT_BINARY_DIR}/include)
 
 if (NOT WITH_SYSTEM_BOOST)
   LIST(APPEND Boost_LIBRARIES "-lz")
+else()
```

Comment:
could you adjust the `if` block a little bit so if looks like
```cmake
if (WITH_SYSTEM_BOOST)
  if (FREEBSD)
    # if boost_python is used then also link with libpython*
    LIST(APPEND Boost_LIBRARIES ${PYTHON_LIBRARIES})
  endif()
else()
  LIST(APPEND Boost_LIBRARIES "-lz")
endif()
```
 because the positive logic is easier to understand.